build-support/bin/native/bootstrap.sh

```diff
@@ -95,17 +103,21 @@ function bootstrap_native_code() {
   # Bootstraps the native code and overwrites the native_engine_version to the resulting hash
   # version if needed.
   local native_engine_version="$(calculate_current_hash)"
```

Comment:
I think this hash should include the native.py file, since it's part of the binary now.

I'd also like to have tests for these bootstrap bits, but I won't block on it.